edb/server/compiler/compiler.py

```diff
@@ -1897,6 +1900,9 @@ def _try_compile(
                     unit.tx_commit = True
                 elif comp.tx_action == dbstate.TxAction.ROLLBACK:
                     unit.tx_rollback = True
+                elif comp.tx_action == dbstate.TxAction.ROLLBACK_TO_SAVEPOINT:
+                    unit.tx_savepoint_rollback = True
+                    unit.sp_name = dbstate.ABORT_MIGRATION_SAVEPOINT
```

Comment:
Like @1st1 suggested, we could make the `tx_commit`, `tx_rollback`, `tx_savepoint_rollback`, `tx_savepoint_commit` (and this hack now) into an enum like this:

```python
class TxCommand(enum.IntEnum):

    COMMIT = 1
    ROLLBACK = 2

    DECLARE_SAVEPOINT = 3
    ROLLBACK_TO_SAVEPOINT = 4

    ABORT_MIGRATION = 5
```

Reply:
Hmm actually I think the `tx_*` booleans are intentionally not an enum so that the I/O server could do fast `if query_unit.tx_commit` without extra enum API from the `edb.server.compiler` module.